src/main/java/org/quiltmc/loader/impl/metadata/qmj/V1ModMetadataReader.java

```diff
@@ -237,6 +238,24 @@ private static V1ModMetadataImpl readFields(JsonLoaderValue.ObjectImpl root) {
 				}
 			}
 
+			@Nullable
+			JsonLoaderValue intermediateMappingsValue = quiltLoader.get("intermediate_mappings");
+
+			if (intermediateMappingsValue != null) {
+				String mappings = string(quiltLoader, "intermediate_mappings");
+
+				if (mappings.equals("hashed")) {
+					throw parseException(intermediateMappingsValue, "Oh no! This version of Quilt Loader doesn't support hashed mappings, please update Quilt Loader to use this mod.");
+				} else if (!mappings.equals("intermediary")) {
+					throw parseException(intermediateMappingsValue, "unknown intermediate mappings");
+				}
+
+				intermediateMappings = mappings;
+			} else {
+				// Fallback to default mappings
+				intermediateMappings = "hashed";
```

Comment:
This needs to happen before checking if the value is hashed, otherwise you're letting through hashed mods that aren't supported!

Reply:
Hmm, I think I was trying to limit the number of checks that had to happen, but you're right. I'll set it up the way it will be when Loader supports hashed and then fail if hashed after.